File: week2/pset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int count_grade(int words, int sentences, int letters);

int main(void)
{
    string prompt = get_string("Text: ");
    int words = count_words(prompt);
    int sentences = count_sentences(prompt);
    int letters = count_letters(prompt);
    int grade = count_grade(words, sentences, letters);
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
    return 0;
}

int count_grade(int words, int sentences, int letters)
{
    double L = 100.0 * letters / words;
    double S = 100.0 * sentences / words;
    return round(0.0588 * L - 0.296 * S - 15.8);
}

int count_letters(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            counter++;
        }
    }
    return counter;
}

int count_words(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            counter++;
        }
    }
    return counter + 1;
}

int count_sentences(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            counter++;
        }
    }
    return counter;
}

File: week6/pset6/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    sentences = count_sentences(text)
    words = count_words(text)
    grade = count_grade(letters, sentences, words)
    if grade >= 15:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    counter = 0
    for i in range(len(text)):
        if text[i].isalpha():
            counter = counter + 1
    return counter


def count_words(text):
    counter = 0
    for i in range(len(text)):
        if text[i].isspace():
            counter = counter + 1
    return counter + 1


def count_sentences(text):
    counter = 0
    for i in range(len(text)):
        if text[i] == "." or text[i] == "?" or text[i] == "!":
            counter = counter + 1
    return counter


def count_grade(letters, sentences, words):
    L = 100.0 * letters / words
    S = 100.0 * sentences / words
    return round(0.0588 * L - 0.296 * S - 15.8)


if __name__ == "__main__":
    main()
